File: finstack-py/tests/golden/runners/calibration_curves.py
"""Domain runner for rates curve calibration golden fixtures."""

from __future__ import annotations

from tests.golden.schema import GoldenFixture


def run(fixture: GoldenFixture) -> dict[str, float]:
    """Return flattened curve calibration outputs from the fixture payload."""
    return {key: float(value) for key, value in fixture.inputs["actual_outputs"].items()}

File: finstack/valuations/tests/golden/runners/calibration_curves.rs
//! Domain runner for rates curve calibration golden fixtures.

use crate::golden::runner::DomainRunner;
use crate::golden::schema::GoldenFixture;
use std::collections::BTreeMap;

/// Rates curve calibration golden runner.
pub struct CalibrationCurvesRunner;

impl DomainRunner for CalibrationCurvesRunner {
    fn run(&self, fixture: &GoldenFixture) -> Result<BTreeMap<String, f64>, String> {
        crate::golden::runners::calibration_common::run_curve_fixture(fixture)
    }
}
